conversions/binary_to_decimal.c

```diff
@@ -1,24 +1,113 @@
-/**
- * Modified 07/12/2017, Kyler Smith
- *
+/** 
+ * @file binary_to_decimal.c
+ * @brief Converts a binary number to a decimal one.
+ * @details
+ * A binary number is input , it is check to be binary
+ * then number is converted to a decimal number.
+ * Some tests are added too.
+ * Modified 07/12/2017
+ * Modified 2021
+ * @author Kyler Smith
+ * @author [lazy-dude] (https://github.com/lazy-dude)
  */
 
+// includes
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h> 
 #include <stdio.h>
+#include <stdlib.h>
+
+// function prototypes
+bool is_binary(intmax_t num);
+int num_len(intmax_t num);
+intmax_t binary_decimal(intmax_t num);
+void test(void);
 
-int main()
+/**
+ * @brief main function
+ * @param void
+ * @returns 0 on exit
+ */
+int main(void)
 {
-    int remainder, number = 0, decimal_number = 0, temp = 1;
-    printf("\n Enter any binary number= ");
-    scanf("%d", &number);
+    test();
+    printf("All tests passed.\n");
+    return 0;
+}
+
+/**
+ * @brief is_binary checks whether num is a binary one
+ * @param num to be checked if it has binary representation
+ * @return boolean true if num is binary false if not
+ */
+bool is_binary(intmax_t num)
+{
+    int remainder = 0;
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Requires adding the `inttypes.h` library. Check other parts of the code ([reference](https://en.cppreference.com/w/c/types/integer)). 🙂

Reply:
uintmax_t has the biggest length for machine running the code. I would prefer it over uint64_t which has a fixed size.